dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -128,186 +141,246 @@ public void run(final JobExecutionContext jobContext) throws JobExecutionExcepti
-            //Or there wasn't any previous index.
-            //it must be done on the last round of our loop.
-            final boolean switchIndex = !incremental && !listIterator.hasNext();
-            config.setSwitchIndexWhenDone(switchIndex);
-            publisherAPI.publish(config, status);
-        }
+            final Builder<SiteSearchConfig> builder = ImmutableList.builder();
+
+            final List<String> languageToIndex = Arrays
+                    .asList((String[]) dataMap.get(LANG_TO_INDEX));
+            final ListIterator<String> listIterator = languageToIndex.listIterator();
+            while (listIterator.hasNext()) {
+                final String lang = listIterator.next();
+                final SiteSearchConfig config = new SiteSearchConfig();
+                config.setJobId(jobId);
+                config.setLanguage(Long.parseLong(lang));
+                config.setJobName(jobName);
+                config.setHosts(hosts);
+                config.setNewIndexName(newIndexName);
+                config.setIndexName(indexName);
+                config.setIndexAlias(indexAlias);
+                config.setId(bundleId);
+                config.setStartDate(startDate);
+                config.setEndDate(endDate);
+                config.setIncremental(incremental);
+                config.setUser(userToRun);
+
+                if (include) {
+                    config.setIncludePatterns(paths);
+                } else {
+                    config.setExcludePatterns(paths);
+                }
 
-        int filesCount = 0, pagesCount = 0, urlmapCount = 0;
-        for (final BundlerStatus bs : status.getBundlerStatuses()) {
-            if (bs.getBundlerClass().equals(FileAssetBundler.class.getName())) {
-                filesCount += bs.getTotal();
-            } else if (bs.getBundlerClass().equals(URLMapBundler.class.getName())) {
-                urlmapCount += bs.getTotal();
-            } else if (bs.getBundlerClass().equals(HTMLPageAsContentBundler.class.getName())) {
-                pagesCount += bs.getTotal();
+                //We should always replace the index when performing on non-incremental mode.
+                //That means we drop the old one and re-use the alias.
+                //But we only activate the new index when the old one was the default.
+                //Or there wasn't any previous index.
+                //it must be done on the last round of our loop.
+                final boolean switchIndex = !incremental && !listIterator.hasNext();
+                config.setSwitchIndexWhenDone(switchIndex);
+                builder.add(config);
             }
+            final String joinedHosts = UtilMethods.join(indexHosts, ",", true);
+            final String langList = UtilMethods.join(languageToIndex, ",");
+            final String pathsAsString = paths.size() > 0 ? UtilMethods.join(paths, ",") : "/*";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5188880)